awx/ui_next/src/screens/Template/Survey/SurveyQuestionAdd.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import { useHistory, useRouteMatch } from 'react-router-dom';
+import { CardBody } from '@components/Card';
+import SurveyQuestionForm from './SurveyQuestionForm';
+
+export default function SurveyQuestionAdd({ survey, updateSurvey }) {
+  const [formError, setFormError] = useState(null);
+  const history = useHistory();
+  const match = useRouteMatch();
+
+  const handleSubmit = async question => {
+    if (survey.spec.find(q => q.variable === question.variable)) {
```

Comment:
Two things:
 - We probably want `some` instead of `find` since the return value doesn't seem to be used.
 - `survey` is undefined in some cases - like when adding a new question to a template without a survey - so I think we need to handle it. When `survey` is undefined, it prevents the form from saving. I'd expect this line to fail with an error of some sort but it doesn't for some reason - the code just stops running at the `if` check and doesn't proceed.

Reply:
fixed the issues here... gonna dig in a bit more and see if I can figure out why it was swallowing the uncaught error

Reply:
>  I'd also expect this line to fail with an error of some sort but it doesn't for some reason

I looked into this more. Possibly out of scope for this PR, but something to be aware of. Try adding a divide by zero to some code inside of the submit handler:

![Screenshot from 2020-03-20 12-44-34](https://user-images.githubusercontent.com/9753817/77185980-9d0ac480-6aa8-11ea-8fed-2e39f8a27ac9.png)

![surveysaveattempt](https://user-images.githubusercontent.com/9753817/77186160-d4797100-6aa8-11ea-81f3-06d49703ad7b.gif)

Reply:
:laughing: ignore the above. just javascript.

Reply:
A better example, using a reference error:

![Screenshot from 2020-03-20 12-50-39](https://user-images.githubusercontent.com/9753817/77186591-8022c100-6aa9-11ea-8004-7a47ed1784c4.png)

![surveysaveattempt](https://user-images.githubusercontent.com/9753817/77186637-94ff5480-6aa9-11ea-8aad-56e1af47cbfa.gif)